.github/workflows/be_review_prs.yml

```diff
@@ -0,0 +1,52 @@
+name: Enforce PR Approval Requirements
```

Comment:
Can we change this name? I'd like to make it abundantly clear why it's failing so there's minimal confusion 😄 Maybe "Require backend-review-group approval" or "Backend-review-group approval required"?
![image](https://github.com/user-attachments/assets/d2623e9c-4c7c-40e6-8bce-30cac80dcb5d)